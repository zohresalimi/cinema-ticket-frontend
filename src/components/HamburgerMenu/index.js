import React from "react";
import styled from "styled-components";
import { useTranslation } from "react-i18next";

const StyledBurger = styled.button`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: ${({ open }) => (open ? "2rem" : "1.5rem")};
  width: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 99;
  display: flex;
  position: relative;

  &:focus {
    outline: none;
  }

  div {
    width: ${({ open }) => (open ? "2rem" : "1.5rem")};
    height: 0.2rem;
    background: ${({ open }) => (open ? "#0D0C1D" : "#FFF")};
    border-radius: 10px;
    transition: all 0.3s linear;
    position: relative;
    transform-origin: 1px;

    :first-child {
      transform: ${({ open }) => (open ? "rotate(45deg)" : "rotate(0)")};
    }

    :nth-child(2) {
      opacity: ${({ open }) => (open ? "0" : "1")};
      transform: ${({ open }) => (open ? "translateX(20px)" : "translateX(0)")};
    }

    :nth-child(3) {
      transform: ${({ open }) => (open ? "rotate(-45deg)" : "rotate(0)")};
    }
  }

  @media only screen and (min-width: 768px) {
    display: none;
  }
`;

const StyledMenu = styled.nav`
  display: flex;
  flex-direction: column;
  justify-content: center;
  background: ${(props) => props.theme.colors.white};
  transform: ${({ open }) => (open ? "translateX(0)" : "translateX(-100%)")};
  height: 100vh;
  text-align: left;
  position: fixed;
  top: 0;
  left: 0;
  transition: transform 0.3s ease-in-out;
  z-index: 9;

  @media (max-width: 576px) {
    width: 100%;
  }

  a {
    font-size: 2rem;
    text-transform: uppercase;
    padding: 2rem 0;
    font-weight: bold;
    letter-spacing: 0.5rem;
    color: ${(props) => props.theme.colors.black};
    text-decoration: none;
    transition: color 0.3s linear;

    @media (max-width: 576px) {
      font-size: 1.5rem;
      text-align: center;
    }

    &:hover {
      color: #343078;
    }
  }
`;
function HamburgerMenu({ open, setOpen }) {
  const { t } = useTranslation();

  return (
    <>
      <Burger open={open} setOpen={setOpen} />
      <StyledMenu open={open}>
        <a href="/">{t("home")}</a>
        <a href="/">{t("movies")}</a>
      </StyledMenu>
    </>
  );
}

export default HamburgerMenu;

const Burger = ({ open, setOpen }) => {
  return (
    <StyledBurger open={open} onClick={() => setOpen(!open)}>
      <div />
      <div />
      <div />
    </StyledBurger>
  );
};
