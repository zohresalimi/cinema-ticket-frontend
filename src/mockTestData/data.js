import React from "react";
import { Router } from "@reach/router";
import AppContext from "../store/context";
import Theme from "../Styles/Theme";

export const getTestStore = () => {
  return {
    dispatch: () => {},
    state: {
      user: {
        name: "",
        email: "",
      },
      movies: {
        children: [
          {
            _id: "5fa71823adc463397a50c501",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "trolls 2: world tour",
            originalTitle: "swedish",
            premiere: "2020-10-03T15:18:52.722Z",
            rooms: ["5fa653d0a7f1201999dd1ac0"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        premiered: [
          {
            _id: "5fa71823adc463397a50c502",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "follow me",
            originalTitle: "swedish",
            premiere: "2020-10-05T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        upcoming: [
          {
            _id: "5fa71823adc463397a50c503",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "honest thief",
            originalTitle: "swedish",
            premiere: "2021-01-03T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        allMovies: {
          "5fa71823adc463397a50c501": "children",
          "5fa71823adc463397a50c502": "premiered",
          "5fa71823adc463397a50c503": "upcoming",
        },
      },
      cinemas: [
        {
          _id: "5fa64fe234fbbf9945f3a3ce",
          name: "filmstaden kista",
          purchaseEndTime: "21:30",
          purchaseStartTime: "8:00",
          rooms: ["5fa653c0643c06586bba7d78", "5fa653d0a7f1201999dd1ac0"],
        },
      ],
      ticket: {
        movie: {},
        status: false,
        quantity: 0,
        price: 0,
        seatNumbers: new Map(),
        user: {},
        showing: {},
      },
      showings: [
        {
          _id: "5fbe9bd31f58462b5c759b5c",
          capacity: 4,
          cinema: "5fa64fe234fbbf9945f3a3ce",
          endTime: null,
          movie: {
            _id: "5fa71823adc463397a50c502",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            name: "follow me",
            originalTitle: "swedish",
            premiere: "2020-10-05T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
          price: 100,
          room: { _id: "5fa653c0643c06586bba7d78" },

          seats: [
            [
              { seatNumber: 1, taken: "available" },
              { seatNumber: 1, taken: "available" },
            ],
            [
              { seatNumber: 1, taken: "available" },
              { seatNumber: 1, taken: "available" },
            ],
          ],
          startTime: "2020-11-13T16:00:00.000Z",
        },
      ],
    },
  };
};

export const WithProvider = (props) => {
  const { state, dispatch } = getTestStore();
  return (
    <Theme>
      <AppContext.Provider
        value={{ state: props.defaultValue || state, dispatch }}
      >
        <Router>{props.children}</Router>
      </AppContext.Provider>
    </Theme>
  );
};
