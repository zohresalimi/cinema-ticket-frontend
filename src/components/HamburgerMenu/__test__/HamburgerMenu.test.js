import React from "react";
import { render, act, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import HamburgerMenu from "../";

import { WithProvider } from "../../../mockTestData/data";

jest.mock("react-i18next", () => ({
  useTranslation: () => [(key) => key, { changeLanguage: jest.fn() }],
}));

function renderWrapper(props) {
  let component;

  act(() => {
    component = render(
      <WithProvider>
        <HamburgerMenu {...props} path="/" />
      </WithProvider>
    );
  });

  return component;
}

describe("Hamburger Menu Component Testing", () => {
  test("take snapshot", async () => {
    const { container } = await renderWrapper();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should show humbergrt menu by click", async () => {
    const mockSetOpenMenu = jest.fn(() => true);
    const { getByTestId } = await renderWrapper({
      toggleMenu: mockSetOpenMenu,
    });
    fireEvent.click(getByTestId("burger-wrapper"));
    expect(mockSetOpenMenu).toHaveBeenCalled();
  });
});
