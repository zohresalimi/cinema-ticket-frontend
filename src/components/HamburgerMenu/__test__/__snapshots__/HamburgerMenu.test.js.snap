// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger Menu Component Testing take snapshot 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <button
    class="HamburgerMenu__StyledBurger-pqm95a-0 fuWDqq"
    data-testid="burger-wrapper"
  >
    <div />
    <div />
    <div />
  </button>
  <nav
    class="HamburgerMenu__StyledMenu-pqm95a-1 dMjmXU"
  >
    <a
      href="/"
    >
      home
    </a>
    <a
      href="/"
    >
      movies
    </a>
    <a
      href="/sign-up"
    >
      sign up
    </a>
  </nav>
</div>
`;
